lib/cli/ui/printer.rb

```diff
@@ -0,0 +1,43 @@
+require 'cli/ui'
+
+module CLI
+  module UI
+    class Printer
+      # Print a message to a stream with common utilities.
+      # Allows overriding the color, encoding, and target stream.
+      # By default, it formats the string using CLI:UI and rescues common stream errors.
+      #
+      # ==== Attributes
+      #
+      # * +msg+ - (required) the string to output. Can be frozen.
+      #
+      # ==== Options
+      #
+      # * +:frame_color+ - Override the frame color. Defaults to nil.
+      # * +:stream+ - Target stream, like $stdout or $stderr. Can be anything with a puts method. Defaults to $stdout.
+      # * +:encoding+ - Force the output to be in a certain encoding. Defaults to UTF-8.
+      # * +:format+ - Whether to format the string using CLI::UI.fmt. Defaults to true.
+      # * +:graceful+ - Whether to gracefully rescue common I/O errors. Defaults to true.
+      #
+      # ==== Example
+      #
+      #   CLI::UI::Printer.puts('{x} Ouch', stream: $stderr, color: :red)
+      #
+      def self.puts(msg, frame_color: nil, stream: $stdout, encoding: Encoding::UTF_8, format: true, graceful: true)
```

Comment:
I'd call `stream:` `to:`

Reply:
So you'd pass `encoding: nil` to prevent the `force_encoding` call. That's... probably reasonable? I wonder if it should just be a boolean, I can't think of a situation where we'd ever print non-UTF-8 test.

This is probably fine as-is.

Reply:
Me neither, but since this library is intended to be rather flexible, I’d rather not force it.